lib/AST/Decl.cpp

```diff
@@ -4286,6 +4286,11 @@ AbstractFunctionDecl::getDefaultArg(unsigned Index) const {
   llvm_unreachable("Invalid parameter index");
 }
 
+inline Type AbstractFunctionDecl::getMethodInterfaceType() const {
+  assert(getDeclContext()->isTypeContext());
+  return getInterfaceType()->castTo<AnyFunctionType>()->getResult();
```

Comment:
One suggestion: this will assert if the method interface type is ErrorType. You should check for hasError(), and just return ErrorType::get() instead.